### PR TITLE
Add custom logo on the start page

### PR DESCRIPTION
**Pull request by lix0st**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/PhotoboothProject/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)
- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Other, please explain:

#### What changes did you make? (Give an overview)
Add the ability to display a logo on the start page.
The user can choose to show the logo or not, specify the location (path - default to private/logo.png) of the logo and its position (one of the four corners).

#### Is there anything you'd like reviewers to focus on?
 - The logo is displayed as is, there is no scaling. If there are suggestion to handle the size of the logo "automatically" it would be great.
  Sadly I'm not a web developer ;)

- I'm not sure I've done the translation the right way. The Contribution documentation tells "translation need to be done on [Crowdin](https://crowdin.com/project/photobooth)", but I don't know how it works. So I just added the required text for EN, FR and DE (the languages that I know).

### CODE REVIEW

**Comment by andi34** [comment]
Thanks a lot! I'll take a closer look next days. French and German translation will be removed on next crowdin sync of they aren't added there too.

https://crowdin.com/project/photobooth

You can apply as German and french translator if you'd like.

File: src/sass/modern_style.scss [hunk 1]
```diff
@@ -72,6 +72,27 @@ hr.small {
     top: 5px;
     bottom: auto;
   }
+
```
Comment (andi34) [inline src/sass/modern_style.scss]: Not needed in this file, we have the `@import 'classic_style';` which will include it. Only needed here if we want to apply a different styling.

**Review by andi34 — CHANGES_REQUESTED** [review]
Changes to the modern_style.scss should be removed.

File: src/sass/modern_style.scss (hunk 1 above)
Comment (lix0st) [reply src/sass/modern_style.scss]: Thanks! Didn't saw the import :)

**Comment by andi34** [comment]
One more thing:
On save, we should validate that the logo exists if enabled and disable it if the file can't be found or if the config is empty. Should be added here: https://github.com/PhotoboothProject/photobooth/blob/dev/api/admin.php

**Comment by lix0st** [comment]
Good point! I've added the validation in the latest commit.

File: api/admin.php [hunk 2]
```diff
@@ -154,6 +154,13 @@
         }
     }
 
+    if ($newConfig['logo']['enabled']) {
+        if (empty($newConfig['logo']['path']) && !file_exists($newConfig['logo']['path'])) {
```
Comment (andi34) [inline api/admin.php]: `!file_exists('..' . DIRECTORY_SEPARATOR . 
 $newConfig['logo']['path'])` ?

File: api/admin.php [hunk 3]
```diff
@@ -154,6 +154,13 @@
         }
     }
 
+    if ($newConfig['logo']['enabled']) {
+        if (empty($newConfig['logo']['path']) && !file_exists('..' . DIRECTORY_SEPARATOR . $newConfig['logo']['path'])) {
+            $newConfig['logo']['enabled'] = false;
+            $LogData[] = ['logo' => 'Logo file path does not exist or is empty. Logo disabled'];
+        }
```
Comment (andi34) [inline api/admin.php:163]:
```php
} else {
    $newConfig['logo']['path'] = fixSeperator($newConfig['logo']['path']);
}
```

Comment (andi34) [reply api/admin.php:163]: We need to take care about using the right seperator

File: resources/lang/de.json [hunk 4]
```diff
@@ -102,6 +102,9 @@
   "frontpage:event_textRight": "Text rechts",
   "frontpage:ui_show_fork": "Fork Logo anzeigen",
   "frontpage:ui_skip_welcome": "Begrüßungsbildschirm überspringen",
+  "frontpage:logo_enabled": "Logo anzeigen",
```
Comment (andi34) [inline resources/lang/de.json:103]: Alphabetic order, must be above `frontpage:ui_`

File: resources/lang/de.json [hunk 5]
```diff
@@ -269,6 +272,9 @@
   "manual:frontpage:event_textRight": "Geben Sie den Text ein, der auf der rechten Seite des ausgewählten Symbols sichtbar ist.",
   "manual:frontpage:ui_show_fork": "Wenn diese Option aktiviert ist, wird in der rechten oberen Ecke des Startbildschirms ein Fork Badge angezeigt.",
   "manual:frontpage:ui_skip_welcome": "Wenn Option aktiviert, ist die Begrüßungsseite beim ersten Zugriff auf Photobooth nach Installation/Update nicht sichtbar.",
+  "manual:frontpage:logo_enabled": "Wenn diese Option aktiviert ist, wird das Logo auf der Startseite angezeigt",
```
Comment (andi34) [inline resources/lang/de.json:273]: Alphabetic order, must be above `manual:frontpage:ui_`

File: resources/lang/en.json [hunk 6]
```diff
@@ -102,6 +102,9 @@
   "frontpage:event_textRight": "Text right",
   "frontpage:ui_show_fork": "Show Fork Badge",
   "frontpage:ui_skip_welcome": "Skip welcome screen",
+  "frontpage:logo_enabled": "Show logo",
```
Comment (andi34) [inline resources/lang/en.json:103]: Alphabetic order applies here too

File: resources/lang/en.json [hunk 7]
```diff
@@ -270,6 +273,9 @@
   "manual:frontpage:event_textRight": "Enter the text visible on the right side of the chosen symbol.",
   "manual:frontpage:ui_show_fork": "If enabled, a fork badge is shown on the right upper corner on the start screen.",
   "manual:frontpage:ui_skip_welcome": "If enabled, the welcome screen won't be visible while accessing Photobooth the first time after installation / update.",
+  "manual:frontpage:logo_enabled": "If enabled, the logo will be shown on the frontpage.",
```
Comment (andi34) [inline resources/lang/en.json:274]: Alphabetic order applies here too

File: resources/lang/fr.json [hunk 8]
```diff
@@ -100,6 +100,9 @@
   "frontpage:event_textRight": "Texte droit",
   "frontpage:ui_show_fork": "Affichez le badge de la page Github",
   "frontpage:ui_skip_welcome": "Passer l'écran d'accueil",
+  "frontpage:logo_enabled": "Afficher logo",
```
Comment (andi34) [inline resources/lang/fr.json:101]: Alphabetic order applies here too

File: resources/lang/fr.json [hunk 9]
```diff
@@ -256,6 +259,9 @@
   "manual:frontpage:event_textRight": "Entrez le texte lisible sur le côté droit du symbole choisi.",
   "manual:frontpage:ui_show_fork": "Si activé, un badge de fork est affiché dans le coin supérieur droit de l'écran de démarrage.",
   "manual:frontpage:ui_skip_welcome": "Si activé, l'écran d'accueil ne sera pas visible lors de l'accès au Photobooth la première fois après l'installation / mise à jour.",
+  "manual:frontpage:logo_enabled": "Si activé, le logo sera visible sur la page d'acceuil.",
```
Comment (andi34) [inline resources/lang/fr.json:260]: Alphabetic order applies here too

File: lib/configsetup.inc.php [hunk 10]
```diff
@@ -360,6 +360,31 @@
...
+        'logo_enabled' => [
+            'view' => 'advanced',
+            'type' => 'checkbox',
+            'name' => 'logo[enabled]',
+            'value' => $config['logo']['enabled'],
+        ],
+        'logo_path' => [
+            'view' => 'expert',
+            'type' => 'input',
+            'placeholder' => 'Logo path',
```
Comment (andi34) [inline lib/configsetup.inc.php]:
```php
'placeholder' => $defaultConfig['logo']['path'],
```

**Review by andi34 — CHANGES_REQUESTED** [review]
Sorry, found some more nitpicks 🙈

**Comment by lix0st** [comment]
> Sorry, found some more nitpicks see_no_evil

No problem. This way I can learn something.
I'll correct those.
Thanks :)

**Review by andi34 — APPROVED** [review]

**Pull request merged**